packages/ketcher-core/src/application/editor/tools/select/SelectBase.ts

```diff
@@ -685,63 +845,105 @@ abstract class SelectBase implements BaseTool {
       return;
     }
 
+    if (!this.firstMonomerPositionBeforeMove) {
+      this.firstMonomerPositionBeforeMove = this.editor.drawingEntitiesManager
+        .selectedMonomers[0]?.position
+        ? new Vec2(
+            this.editor.drawingEntitiesManager.selectedMonomers[0]?.position,
+          )
+        : undefined;
+    }
+
+    const firstMonomerPosition =
+      this.editor.drawingEntitiesManager.selectedMonomers[0]?.position;
+    const distanceBetweenFirstMonomerAndCursorBeforeMove =
+      this.firstMonomerPositionBeforeMove &&
+      Vec2.diff(
+        Coordinates.modelToCanvas(this.firstMonomerPositionBeforeMove),
+        this.mousePositionBeforeMove,
+      );
+    const distanceBetweenFirstMonomerAndCursor =
+      firstMonomerPosition &&
+      Vec2.diff(
+        Coordinates.modelToCanvas(firstMonomerPosition),
+        this.editor.lastCursorPositionOfCanvas,
+      );
+    const firstMonomerPositionDelta =
+      distanceBetweenFirstMonomerAndCursorBeforeMove &&
+      distanceBetweenFirstMonomerAndCursor &&
+      Vec2.diff(
+        distanceBetweenFirstMonomerAndCursorBeforeMove,
+        distanceBetweenFirstMonomerAndCursor,
+      );
+    const movementDelta = Coordinates.canvasToModel(
+      firstMonomerPositionDelta ||
+        new Vec2(
+          this.editor.lastCursorPositionOfCanvas.x -
+            this.mousePositionAfterMove.x,
+          this.editor.lastCursorPositionOfCanvas.y -
+            this.mousePositionAfterMove.y,
+        ),
+    );
+
     const modelChanges = new Command();
 
-    const snapResult = this.tryToSnap(event);
+    const snapResult = this.tryToSnap(event, movementDelta);
     const { snapPosition } = snapResult;
 
+    this.editor.transientDrawingView.hideBondSnap();
+    this.editor.transientDrawingView.hideAngleSnap();
+    this.editor.transientDrawingView.hideDistanceSnap();
+    this.editor.transientDrawingView.hideGroupCenterSnap();
```

Comment:
Perhaps we could introduce the cleanup function for TransientDrawingView which will call all of these (and maybe some additional snapping in future)